HM_9/tasks/decorators.py

```diff
@@ -0,0 +1,40 @@
+import csv
+import json
+import os
+from functools import wraps
+
+# * Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+def save_params(func):
+    func_name = func.__name__
+    if os.path.exists(f'out/{func_name}.json'):
+        with open(f'out/{func_name}.json', 'r', encoding='utf-8') as j:
+            result_list = json.load(j)
+    else:
+        result_list = []
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        func_res = func(*args, **kwargs)
+        result = {'args': args,
+                  'kwargs': kwargs,
+                  'result': func_res}
+        result_list.append(result)
+        with open(f'out/{func_name}.json', "w", encoding="utf-8") as j:
+            json.dump(result_list, j, indent=2)
+        return func_res
+
+    return wrapper
+
+# * Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+def calculate(path):
+    def deco(func):
+        results = []
+        def wrapper(*args, **kwargs):
+            with open(path, 'r', encoding='utf-8', newline='') as c:
+                csv_reader = csv.reader(c)
+                for i, line in enumerate(csv_reader):
+                    if i:
+                        results.append(func(*map(int, line), **kwargs))
+            return results
```

Comment:
Помните, я говорил на семинаре, что функция обертка должна как бы "мимикрировать" под оригинальную функцию, то есть она должна принимать и возвращать то же самое, что и декорируемая функция. Но у вас обертка возвращает список. Зачем и почему? Причем в этом даже нет смысла, так как порядок декорирования у вас уже определяется сохранение. Зачем при этом еще помещать результаты работы в список, мне совершенно непонятно.
```suggestion
        def wrapper(*args, **kwargs):
            with open(path, 'r', encoding='utf-8', newline='') as c:
                csv_reader = csv.reader(c)
                for i, line in enumerate(csv_reader):
                    if i:
                        func(*map(int, line))
```